test/testAnnoutil.pl

```diff
@@ -0,0 +1,2223 @@
+my $obj_65386_2 = '65386/12/1';
+
+my $asmb_fasta = $annoutil->_get_fasta_from_assembly($obj_asmb);
+
+#####RAST_SDK Module test objects #####
+my $obj2_1 = "63171/315/1";
+
+# used for function lookup in report testing
+my $test_ftrs = [{
+ 'id' => '10000_1',
+ 'protein_translation' => 'VVVLLHGGCCEDMQRGRRESAPDLTLVVYPHALHALDMRLPDRTVLGMRLGFDAHAAADARRQVLDFLTARGVAPPDR*'
+ },
+ {
+ 'function' => 'L-carnitine dehydratase/bile acid-inducible protein F (EC 2.8.3.16)',
+ 'quality' => {
+ 'hit_count' => 3
+ },
+ 'annotations' => [
+ [
+ 'Function updated to L-carnitine dehydratase/bile acid-inducible protein F (EC 2.8.3.16)',
+ 'annotate_proteins_kmer_v1',
+ '1583389302.95393',
+ '6c670c83-2a11-49ff-97bf-b1c3e2121f30'
+ ]
+ ],
+ 'id' => '10000_2'
+ },
+ {
+ 'id' => '10000_3',
+ 'protein_translation' => 'MLAVNEPTVVLASAETKSLGPVVTGDRLETEAEVERTDGRKRWVKVTVRRAGAPVMEGQFLAVVPDRHILDAKDARR*'
+ },
+ {
+ 'id' => '10000_madeup',
+ 'function' => 'completely fake function',
+ 'annotations' => [
+ [
+ 'completely fake function',
+ 'annotate_madeup_source',
+ '1583389302.95394',
+ '6c670c83-2a11-49ff-97bf-b1c3e2121f33'
+ ]
+ ],
+ }];
+
+
+=begin
+## Re-mapping the contigIDs back to their original (long) names
+subtest '_remap_contigIDs' => sub {
+    my $contigID_hash = {
+          'contigID_1' => 'NZ_CP028859.1',
+          'contigID_2' => 'NZ_CP028860.1'
+    };
+    my $gn = {
+        contigs => [
+            {id => 'contigID_1',
+             name => 'contigID_1'},
+            {id => 'contigID_2',
+             name => 'contigID_2'}
+        ]
+    };
+    $gn = $annoutil->_remap_contigIDs($contigID_hash, $gn);
+    my $contig1 = $gn->{contigs}[0];
+    my $contig2 = $gn->{contigs}[1];
+    ok ($contig1->{id} eq $contigID_hash->{contigID_1}, "mapped contigID correctly");
+    ok ($contig2->{id} eq $contigID_hash->{contigID_2}, "mapped contigID correctly");
+};
+
+subtest '_get_contigs_from_fastafile' => sub {
+    my $fa_file = $fa_LOng;
+    my $contigs;
+    my $contigID_hash;
+    lives_ok {
+        ($contigs, $contigID_hash) = $annoutil->_get_contigs_from_fastafile($fa_file);
+    } '_get_contigs_from_fastafile runs successfully';
+    is ($contigID_hash->{contigID_1},
+        'CP0035411Candidatus_Carsonella_ruddii_CE_isolate_Thao2000_complete_genome',
+        'contigID hash map generated correctly');
+    ok ($contigs->[0]->{id} eq 'contigID_1', 'contig id correctly shortened for downstream');
+    ok ($contigs->[0]->{name} eq 'contigID_1', 'contig name correctly shortened for downstream');
+};
+
+subtest '_get_genome' => sub {
+    my $parameters = {
+         output_genome_name => 'test_out_gn_name',
+         output_workspace => $ws,
+         object_ref => $obj_Ecoli
+    };
+
+    my $obj_ref = $parameters->{object_ref};
+    my $ret_gn;
+    lives_ok { 
+        $ret_gn = $annoutil->_get_genome($obj_ref);
+        print "genome object returned on $obj_ref:\n".Dumper(keys %$ret_gn);
+    } '_get_genome runs successfully';
```

Comment:
You can use Test::Deep's `cmp_deeply` to test the returned data structure more thoroughly. It is much better to write automated tests of the returned data structure rather than relying on printing out the results and looking through them by eye.